### ISSUE TITLE
Cancelling payments puts your unpaid registration in everyone's carts

### ISSUE THREAD

**Issue by adebus** [MidwestFurryFandom/rams#393]
During testing, Selene noted that while Emojis are prohibited from entry in the badge name, they are still valid in other fields. In testing how that works, she did the following:

> Created attendee registration in http://rams.test.furfest.org/
> -Was able to add and save 2 heart emojis to the Legal Name field.
> -Upon check out, added another registration.
> 

At this time, she was able to hit Edit and see the two hearts in the Legal Name field.

Following that, and recognizing that she created a second registration, testing another common symbol:

> 2. Created second registration - put % in the Last Name field
> - Saved and moved to check out.
> - % symbol was seen in the last name of the second attendee (See Screenshot)
> 

![image](https://user-images.githubusercontent.com/4514884/132428768-1e81cc0f-5928-4a11-9867-5341e2272ba7.png)

Simultaneously, I was attempting to create a registration using Chrome. I attempted to create a registration using the email address test123@tauvix.com:

![image](https://user-images.githubusercontent.com/4514884/132429547-13c23a73-7b40-4981-824b-55a89c0b6d86.png)

After clicking "Add to Cart" it took me to the registration account page for testpersonfive@example.com rather than for test123@tauvix.com, despite not being on the same computer or browser session as Selene. I loaded up a secondary browser (Edge) and was able to reproduce the issue there from a new web session. 

Additional from Selene:

> After you came across your issue on your side, I tried to edit on my side and was returned to an empty form with a notification that the registration was finalized. 
> 
> After checking the Admin side of RAMS, neither registration was found. 

![image](https://user-images.githubusercontent.com/4514884/132429910-eb440a24-873a-4b59-bdba-270946038844.png)

Checking again from a new browser (Firefox) about an hour later, I was able to start a new registration, and the account system workflow seems to be functioning in that browser, however, in my original browser with the "not my" cart open, I can still reproduce the issues.

However, closing out previous browsers, reopening them, and trying to register shows appropriate behavior. 

In summary:

On one computer Selene created a user with a '%' character in their name, and once that was done, all attempted registrations on my part showed me her cart - with my registration lost - and it was impossible to edit any of the registrations in that cart. This appeared to happen to anyone who was actively working on a registration at the time this occurred had their sessions combined/invalidated and allowed some minimal amount of PII leakage.

**Comment by kitsuta** [comment]
This implies there's a bug inside cherrypy's session object, which is what we use to manage pending preregistrations. I don't know how deep that particular rabbit hole goes.

Were you able to reproduce this bug with the same steps?
